homeassistant/components/xiaomi_miio/__init__.py

```diff
@@ -88,11 +89,15 @@ async def async_setup_gateway_entry(
 
     async def async_update_data():
         """Fetch data from the subdevice."""
-        try:
-            for sub_device in gateway.gateway_device.devices.values():
+        data = {}
+        for sub_device in gateway.gateway_device.devices.values():
+            try:
                 await hass.async_add_executor_job(sub_device.update)
-        except GatewayException as ex:
-            raise UpdateFailed("Got exception while fetching the state") from ex
+                data[sub_device.sid] = {ATTR_AVAILABLE: True}
```

Comment:
Please just wrap the line that can raise in the try... except block. Use an `else:` block if needed.

Reply:
done